tapestry.js

```diff
@@ -975,39 +1031,121 @@ function getBoundedCoord(coord, maxCoord) {
     return Math.max(MAX_RADIUS, Math.min(maxCoord - MAX_RADIUS, coord));
 }
 
-function getChildren(id) {
-    var children = [];
-    var dataLinks = dataset.links;
-    for (var linkId in dataLinks) {
-        var link = dataLinks[linkId];
-        //SEARCH FOR: CURRENT_ID_NODE -> CHILD links
-        if (typeof link.source === 'number' && link.source === id) {
-            children.push(link.target);
-        } else if (typeof link.source === 'object' && link.source.id === id) {
-            children.push(link.target.id);
-        }
+/* Add 'depth' parameter to each node recursively. 
+   The depth is determined by the number of levels from the root each node is. */
+
+function addDepthToNodes(id, depth, visited) {
+    visited.push(id);
+
+    var depthAt = 0;
+
+    dataset.nodes[findNodeIndex(id)].depth = depth;
+    var children = getChildren(id, 1);
 
-        //SEARCH FOR: PARENT -> CURRENT_ID_NODE links;
-        // These should also appear as "children" because they're adjacent to the current node
-        if (typeof link.target === 'number' && link.target === id) {
-            children.push(link.source);
-        } else if (typeof link.target === 'object' && link.target.id === id) {
-            children.push(link.source.id);
+    var childLevel;
+
+    // progress through every child at a given node one at a time:
+    while (depthAt < children.length) {
+        for (var childId in children) {
+            // if the child has been visited, check to make sure the calculated depth
+            // is as low as it can be (via childLevel) to correct for shorter paths
+            // to the same node.
+            if (visited.includes(children[childId])) {
+                childLevel = depth;
+                if (dataset.nodes[findNodeIndex(children[childId])].depth > childLevel) {
+                    dataset.nodes[findNodeIndex(children[childId])].depth = childLevel;
+                }
+                else {
+                    depthAt++;
+                }
+            }
+            // if the child has not been visited, record its depth (one away from the
+            // current node's childLevel), and recursively add depth to all of the 
+            // child's children.
+            else {
+                childLevel = depth + 1;
+                dataset.nodes[findNodeIndex(children[childId])].depth = childLevel;
+                visited.push(children[childId]);
+
+                addDepthToNodes(children[childId], childLevel, visited);
+                depthAt++;
+            }
         }
     }
+    
+}
+
+/* Return the distance between a node and its farthest descendant node. */
+
+function findMaxDepth(id) {
+
+    // create the .depth parameter for every node
+    addDepthToNodes(id, 0, []);
+    var nodes = dataset.nodes;
+
+    // idList: collect node IDs since they're numbered dynamically
+    var idList = [];
+    var count;
+    for (count = 0; count < nodes.length; count++) {
```

Comment:
If you are only using the count variable within the for loop, you could just declare it within your for loop like 
```
for (var count = 0; count < ...
```

Reply:
corrected on line 1089 just as you said